client/db/bolt/db_test.go

```diff
@@ -509,32 +511,158 @@ func TestOrders(t *testing.T) {
 	if err != nil {
 		t.Fatalf("error setting order status: %v", err)
 	}
-	mord, _ = boltdb.Order(activeOrder.ID())
+	mord, err = boltdb.Order(activeOrder.ID())
 	if mord.MetaData.Status != order.OrderStatusExecuted {
 		t.Fatalf("failed to update order status")
 	}
+	if err != nil {
+		t.Fatalf("failed to load order: %v", err)
+	}
 	// random id should be an error
 	err = boltdb.UpdateOrderStatus(ordertest.RandomOrderID(), order.OrderStatusExecuted)
 	if err == nil {
 		t.Fatalf("no error encountered for updating unknown order's status")
 	}
+}
+
+func TestOrderChange(t *testing.T) {
+	boltdb := newTestDB(t)
+	// Create an account to use.
+	acct := dbtest.RandomAccountInfo()
+	err := boltdb.CreateAccount(acct)
+	if err != nil {
+		t.Fatalf("CreateAccount error: %v", err)
+	}
+
+	ord := randOrderForMarket(randU32(), randU32())
+	mordIn := &db.MetaOrder{
+		MetaData: &db.OrderMetaData{
+			Status: 3,
+			Host:   acct.Host,
+			Proof:  db.OrderProof{DEXSig: randBytes(73)},
+		},
+		Order: ord,
+	}
+	// fmt.Printf("%#v\n", mordIn.MetaData.ChangeCoin) // order.CoinID(nil)
+
+	err = boltdb.UpdateOrder(mordIn)
+	if err != nil {
+		t.Fatalf("error inserting order: %v", err)
+	}
+
+	mord, err := boltdb.Order(ord.ID())
+	if err != nil {
+		t.Fatalf("unable to retrieve order by id")
+	}
+	if mord.MetaData.ChangeCoin != nil {
+		t.Errorf("ChangeCoin was not nil: %#v", mord.MetaData.ChangeCoin)
+	}
 
-	// Set the change coin.
-	err = boltdb.SetChangeCoin(activeOrder.ID(), []byte("abc"))
+	// non-nil empty loads as nil too
+	err = boltdb.SetChangeCoin(ord.ID(), []byte{})
 	if err != nil {
 		t.Fatalf("error setting change coin: %v", err)
 	}
-	mord, _ = boltdb.Order(activeOrder.ID())
-	if string(mord.MetaData.ChangeCoin) != "abc" {
-		t.Fatalf("failed to set change coin")
+	mord, err = boltdb.Order(ord.ID())
+	if err != nil {
+		t.Fatalf("failed to load order: %v", err)
+	}
+	if mord.MetaData.ChangeCoin != nil {
+		t.Fatalf("failed to set change coin, got %#v, want <nil>", mord.MetaData.ChangeCoin)
+	}
+
+	// now some data
+	someChange := []byte{1, 2, 3}
+	err = boltdb.SetChangeCoin(ord.ID(), someChange)
+	if err != nil {
+		t.Fatalf("error setting change coin: %v", err)
+	}
+	mord, err = boltdb.Order(ord.ID())
+	if err != nil {
+		t.Fatalf("failed to load order: %v", err)
+	}
+	if !bytes.Equal(someChange, mord.MetaData.ChangeCoin) {
+		t.Fatalf("failed to set change coin, got %#v, want %#v", mord.MetaData.ChangeCoin, someChange)
 	}
+
 	// random id should be an error
 	err = boltdb.SetChangeCoin(ordertest.RandomOrderID(), []byte("abc"))
 	if err == nil {
 		t.Fatalf("no error encountered for updating unknown order change coin")
 	}
 }
 
+/*
+func TestNilValue(t *testing.T) {
+	tDir, _ := ioutil.TempDir("", "dbtest")
+	defer os.RemoveAll(tDir)
```

Comment:
If you want to leave this comment, let's drop a link to [this convo](https://github.com/decred/dcrdex/pull/531#discussion_r453895125) and a short explanation.

Reply:
It's in a play now instead: go.etcd.io/bbolt